class_name DropPod
extends Node2D

signal opened

@export var velocity_min: Vector2i
@export var velocity_max: Vector2i
@export var angular_veclocity_min: int
@export var angular_veclocity_max: int

@onready var _left_shell: RigidBody2D = $LeftShell
@onready var _right_shell: RigidBody2D = $RightShell


func _ready():
	pass


func land():
	Events.spawn_effect.emit("shatter", global_position + Vector2(0, 8))
	Events.camera_shake.emit(4.0)


func open():
	var y_velocity = randf_range(velocity_min.y, velocity_max.y)
	var x_velocity = randf_range(velocity_min.x, velocity_max.x)
	var angular_veclocity = randf_range(angular_veclocity_min, angular_veclocity_max)

	_left_shell.linear_velocity = Vector2(-x_velocity, -y_velocity)
	_left_shell.angular_velocity = -angular_veclocity
	_left_shell.constant_force = Vector2(-x_velocity / 10, -y_velocity)

	_right_shell.linear_velocity = Vector2(x_velocity, -y_velocity)
	_right_shell.angular_velocity = angular_veclocity
	_right_shell.constant_force = Vector2(x_velocity / 10, -y_velocity)

	Events.spawn_effect.emit("smoke_explosive", global_position)
	Events.camera_shake.emit(2.0)

	opened.emit()
